### PR TITLE
Adds a solgov variant of the squad vendor

### PR DESCRIPTION
**Pull request by Pockets-byte**
it gives out laser pistols not handguns. 
## Why It's Good For The Game
pew

this also allows for not just weapon differences, but multiple squad vendors dispensing different sets of gear on the same map/server, as well as the ability to have varying loadouts, ghost role squad vendors, syndicate squad vendors, etc. 
## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/BeeStation/NSV13/assets/55299415/31986c7e-d69c-44e0-833d-a27a57f836f8

</details>

## Changelog
:cl:
add: Added solgov variant of the squad vendor
/:cl:

### CODE REVIEW

**Review by Bokkiewokkie — CHANGES_REQUESTED** [review]
I don't think we really need to give squad vendors laser guns for Solgov, the glocks are already pretty good and you *can* grab guns from the armory too.

File: nsv13/code/modules/squads/squad_vendor.dm [hunk 1]
```diff
@@ -41,6 +41,15 @@
...
 
+/obj/machinery/squad_vendor/solgov
+	name = "SolGov Squad Vendor"
+
+/obj/machinery/squad_vendor/solgov/Initialize(mapload)
+	. = ..()
+	if(!length(loadouts))
+		for(var/instance in subtypesof(/datum/squad_loadout_solgov))
+			loadouts += new instance
+
```
[the remark this answers is not in the document]
Comment (Pockets-byte) [reply nsv13/code/modules/squads/squad_vendor.dm:52]: oh I see, thanks

[the remark this answers is not in the document]
Comment (Pockets-byte) [reply nsv13/code/modules/squads/squad_vendor.dm]: I have no idea if this works. I'll have to try testing it on my windows copy once I boot into it

**Comment by Pockets-byte** [comment]
> I don't think we really need to give squad vendors laser guns for Solgov, the glocks are already pretty good and you _can_ grab guns from the armory too.

it's less about good, and more about thematics- I'd personally like to see fewer ballistics on solgov ships to make them feel more different- if we need a "less powerful" standard laser pistol to go into this, that's probably an easy shake

File: nsv13/code/modules/squads/squad_vendor.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (Pockets-byte) [reply nsv13/code/modules/squads/squad_vendor.dm]: fixed

**Comment by Pockets-byte** [comment]
if the standard variant of retro lasers are considered to be too strong, they can be exchanged for the old variant

File: nsv13/code/modules/squads/squad_vendor.dm [hunk 2]
```diff
@@ -32,13 +32,19 @@
...
 	req_one_access = list(ACCESS_HOP, ACCESS_HOS)
 	var/static/list/loans_info = list()
-	var/static/list/loadouts = list()
+	var/loadout_type = /datum/squad_loadout/nt
+	var/list/loadouts = list()
 	var/static/list/requires_weapons_clearance = list(/obj/item/ammo_box, /obj/item/gun)
 
+/obj/machinery/squad_vendor/solgov
+	name = "Squad Vendor"
+	icon_state = "squadvend"
```
Comment (Bokkiewokkie) [inline nsv13/code/modules/squads/squad_vendor.dm:41]:
```suggestion
```
These are inherited from the main type

File: nsv13/code/modules/squads/squad_vendor.dm [hunk 3]
```diff
@@ -32,13 +32,19 @@
 	resistance_flags = ACID_PROOF | FIRE_PROOF
```
Comment (Bokkiewokkie) [inline nsv13/code/modules/squads/squad_vendor.dm:32]: Could you make the name of the regular squad vendor written without capitalization? Just an extra request.

**Review by Bokkiewokkie — CHANGES_REQUESTED** [review]

File: nsv13/code/modules/squads/squad_vendor.dm (hunk 2 above)
Comment (Pockets-byte) [reply nsv13/code/modules/squads/squad_vendor.dm:41]: I was thinking about doing a sprite, but I think I'm going to leave it as is for now, so I'll remove it

**Review by Bokkiewokkie — APPROVED** [review]

**Pull request merged**